# DP: Description: Support libraries that has execstack on <= 2.4.18 kernels.
# DP: Related bugs: #321718
# DP: Dpatch author: GOTO Masanori <gotom@debian.org>
# DP: Patch author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Not submitted
# DP: Status Details: Will be submitted
# DP: Date: 2005-08-19

2005-08-19  GOTO Masanori  <gotom@debian.org>

	* sysdeps/unix/sysv/linux/dl-execstack.c: Check not only ENOMEM but
	also EFAULT for pre 2.4.19 kernels.

Index: b/sysdeps/unix/sysv/linux/dl-execstack.c
===================================================================
--- a/sysdeps/unix/sysv/linux/dl-execstack.c	2008-08-08 19:08:18.000000000 +0200
+++ b/sysdeps/unix/sysv/linux/dl-execstack.c	2008-08-08 19:09:03.000000000 +0200
@@ -89,7 +89,7 @@
 	page -= size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EFAULT)	/* Unexpected failure mode.  */
 	    {
 	      result = errno;
 	      goto out;
@@ -115,7 +115,7 @@
 	page += size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EFAULT)	/* Unexpected failure mode.  */
 	    {
 	      result = errno;
 	      goto out;
